### ISSUE TITLE
The type 'array<mixed, mixed>' is more general than the declared return type 'array<string|int, mixed>'

### ISSUE THREAD

**Issue by TysonAndre** [vimeo/psalm#637]
Expected: no warning (Array keys can only be integers or strings, so they're equivalent for most purposes)
Observed: LessSpecificReturnStatement warning

Low priority

**Comment by muglug** [comment]
Reproducible  with

```php
/** @return array<int|string, mixed> */
function returnsArray(array $arr) : array {
  return $arr;
}
```

I agree that *if* the user adds this type annotation themselves, Psalm should not complain.
